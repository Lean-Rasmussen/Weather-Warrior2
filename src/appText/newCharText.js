export const Nerd =' Here is some text about the nerd class';
export const defaultClassText ='This is going to say something about the class you pick';
export const girlScout ='I sell cookies and wear rat tailes and carry a knife';
export const shovelKid ='I once digged all the way to the bottom of the sand bit, guess what it becomes dirt';
export const FatKid ='Guys you gotta wait up, I got honney all over my knees!'
export const Collector ='I got like the best yogiho cards and I caught all the pokermons, so now there is like nothing for me to collect';