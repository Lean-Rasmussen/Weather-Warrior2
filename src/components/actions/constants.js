export const SET_SCREEN ='SET_MAINSCREEN';