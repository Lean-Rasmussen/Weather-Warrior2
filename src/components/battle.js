import React,{ Component} from 'react';

class Battle extends Component{

    render(){
        return(
         <h1>Battle screen</h1>   
        )
    }
}

export default Battle;