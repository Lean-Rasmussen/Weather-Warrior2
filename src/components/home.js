//import modules
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from'redux';

//import components
import PlayerStats from './home/PlayerStats'

export class Home extends Component{
    render(){
        return(
            <div>
                <h1> Home Screen</h1> 
                <PlayerStats data ={40} />
            </div>

           
            
        )
    }
}
export default Home;