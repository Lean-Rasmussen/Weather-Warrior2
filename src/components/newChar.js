import React, { Component } from 'react';

class NewChar extends Component{
    render(){
        return(
            <h1>New Char</h1>
        )
    }
}
export default NewChar;