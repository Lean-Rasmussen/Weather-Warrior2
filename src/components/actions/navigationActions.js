import * as constants from './constants';

export const navAction = (screen)=>(
    {
        type: constants.SET_SCREEN,
        mainScreen: screen,
    }

)