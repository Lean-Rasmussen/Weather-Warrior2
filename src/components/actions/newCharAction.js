import * as constants from '../actions/constants';

export const classSelected = (SelectedClass)=>(
    {
        type: constants.CLASS_SELECTED,
        selectedclass: SelectedClass,
    }
)

export const inputName = (name)=>({
    type: constants.INPUT_NAME,
    name: name,
    }
)