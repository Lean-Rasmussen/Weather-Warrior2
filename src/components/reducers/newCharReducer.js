import * as constants from '../actions/constants'

const newCharReducer =(state=  '', action)=>{
    switch(action.type){
        //class selection
        case constants.CLASS_SELECTED:
            return action.selectedclass;
        //Name Input 
        case constants.INPUT_NAME:
            return action.name;
        // returning on default    
        default:
            return state
    }
}

export default newCharReducer;