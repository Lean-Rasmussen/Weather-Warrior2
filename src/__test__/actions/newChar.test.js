import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import {connect} from 'react-redux';
import * as constants from '../../components/actions/constants';
import * as newCharAction from '../../components/actions/newCharAction';
import * as text from '../../appText/newCharText'
import newCharReducer from '../../components/reducers/newCharReducer'

import {NewChar} from '../../components/newChar'

describe('test of new char container',()=>{
    const newChar = shallow(<NewChar />);

    it('renders like normal snapshot test',()=>{
        expect(newChar).toMatchSnapshot();
    })
    it('Selector Action Class is returned when selected',()=>{
        const mockClassSelected = 'Nerd';
        const expectedActionReturn = {
            type: constants.CLASS_SELECTED,
            selectedclass: mockClassSelected,
        };
        expect(newCharAction.classSelected(mockClassSelected)).toEqual(expectedActionReturn)
    })
    it('Selector Reducer reducer returns informaiton on class and sets it to state',()=>{
        const mockClassSelected = 'Nerd';
        let testAction = newCharAction.classSelected(mockClassSelected)
        expect(newCharReducer( undefined, testAction ))
        .toEqual(mockClassSelected)
    })
    it('InputName Action sends action correctly when changing the input field',()=>{
        const mockName= 'Johnathan'
        const expectedActionReturn = {
            type: constants.INPUT_NAME,
            name: mockName,
        }
        expect(newCharAction.inputName(mockName)).toEqual(expectedActionReturn)
    })
    it('InputName Reducer sends Name to the state via reducer',()=>{
        const mockName ='Bente Bongo';
        let testAction = newCharAction.inputName( mockName);
        expect(newCharReducer(undefined, testAction)).toEqual(mockName)
    })
})
