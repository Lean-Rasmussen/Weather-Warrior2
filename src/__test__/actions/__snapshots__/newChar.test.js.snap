// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of new char container renders like normal snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewChar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1>
                    Make a new Guy
          </h1>
</div>,
        <div>
          <p>
                     Name: 
          </p>
          <input
                    class="NewChar__input"
                    onCHange={[Function]}
          />
          <p>
                     Class 
          </p>
          <select
                    onChange={[Function]}
          >
                    <option>
                               Pick a class 
                    </option>
                    <option
                              id="Nerd"
                    >
                               Nerd 
                    </option>
                    <option
                              id="Girl Scout"
                    >
                               Girl Scout 
                    </option>
                    <option
                              id="Shovel kid"
                    >
                               Shovel Kid
                    </option>
                    <option
                              id="Collector"
                    >
                               Collector
                    </option>
                    <option
                              id="Fat Kid"
                    >
                               Fat Kid
                    </option>
          </select>
          <div>
                    <p>
                              This is going to say something about the class you pick
                    </p>
          </div>
          <button>
                     OK yea I made my guy 
          </button>
</div>,
        <div>
           here goes your hero gif
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Make a new Guy
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Make a new Guy",
          },
          "ref": null,
          "rendered": "Make a new Guy",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
               Name: 
</p>,
            <input
              class="NewChar__input"
              onCHange={[Function]}
/>,
            <p>
               Class 
</p>,
            <select
              onChange={[Function]}
>
              <option>
                             Pick a class 
              </option>
              <option
                            id="Nerd"
              >
                             Nerd 
              </option>
              <option
                            id="Girl Scout"
              >
                             Girl Scout 
              </option>
              <option
                            id="Shovel kid"
              >
                             Shovel Kid
              </option>
              <option
                            id="Collector"
              >
                             Collector
              </option>
              <option
                            id="Fat Kid"
              >
                             Fat Kid
              </option>
</select>,
            <div>
              <p>
                            This is going to say something about the class you pick
              </p>
</div>,
            <button>
               OK yea I made my guy 
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Name: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              " Name: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "class": "NewChar__input",
              "onCHange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Class ",
            },
            "ref": null,
            "rendered": " Class ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option>
                   Pick a class 
</option>,
                <option
                  id="Nerd"
>
                   Nerd 
</option>,
                <option
                  id="Girl Scout"
>
                   Girl Scout 
</option>,
                <option
                  id="Shovel kid"
>
                   Shovel Kid
</option>,
                <option
                  id="Collector"
>
                   Collector
</option>,
                <option
                  id="Fat Kid"
>
                   Fat Kid
</option>,
              ],
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Pick a class ",
                },
                "ref": null,
                "rendered": " Pick a class ",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Nerd ",
                  "id": "Nerd",
                },
                "ref": null,
                "rendered": " Nerd ",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Girl Scout ",
                  "id": "Girl Scout",
                },
                "ref": null,
                "rendered": " Girl Scout ",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Shovel Kid",
                  "id": "Shovel kid",
                },
                "ref": null,
                "rendered": " Shovel Kid",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Collector",
                  "id": "Collector",
                },
                "ref": null,
                "rendered": " Collector",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Fat Kid",
                  "id": "Fat Kid",
                },
                "ref": null,
                "rendered": " Fat Kid",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                This is going to say something about the class you pick
</p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is going to say something about the class you pick",
              },
              "ref": null,
              "rendered": "This is going to say something about the class you pick",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " OK yea I made my guy ",
            },
            "ref": null,
            "rendered": " OK yea I made my guy ",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " here goes your hero gif",
        },
        "ref": null,
        "rendered": " here goes your hero gif",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1>
                        Make a new Guy
            </h1>
</div>,
          <div>
            <p>
                         Name: 
            </p>
            <input
                        class="NewChar__input"
                        onCHange={[Function]}
            />
            <p>
                         Class 
            </p>
            <select
                        onChange={[Function]}
            >
                        <option>
                                     Pick a class 
                        </option>
                        <option
                                    id="Nerd"
                        >
                                     Nerd 
                        </option>
                        <option
                                    id="Girl Scout"
                        >
                                     Girl Scout 
                        </option>
                        <option
                                    id="Shovel kid"
                        >
                                     Shovel Kid
                        </option>
                        <option
                                    id="Collector"
                        >
                                     Collector
                        </option>
                        <option
                                    id="Fat Kid"
                        >
                                     Fat Kid
                        </option>
            </select>
            <div>
                        <p>
                                    This is going to say something about the class you pick
                        </p>
            </div>
            <button>
                         OK yea I made my guy 
            </button>
</div>,
          <div>
             here goes your hero gif
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Make a new Guy
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Make a new Guy",
            },
            "ref": null,
            "rendered": "Make a new Guy",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                 Name: 
</p>,
              <input
                class="NewChar__input"
                onCHange={[Function]}
/>,
              <p>
                 Class 
</p>,
              <select
                onChange={[Function]}
>
                <option>
                                 Pick a class 
                </option>
                <option
                                id="Nerd"
                >
                                 Nerd 
                </option>
                <option
                                id="Girl Scout"
                >
                                 Girl Scout 
                </option>
                <option
                                id="Shovel kid"
                >
                                 Shovel Kid
                </option>
                <option
                                id="Collector"
                >
                                 Collector
                </option>
                <option
                                id="Fat Kid"
                >
                                 Fat Kid
                </option>
</select>,
              <div>
                <p>
                                This is going to say something about the class you pick
                </p>
</div>,
              <button>
                 OK yea I made my guy 
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Name: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                " Name: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "class": "NewChar__input",
                "onCHange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Class ",
              },
              "ref": null,
              "rendered": " Class ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option>
                     Pick a class 
</option>,
                  <option
                    id="Nerd"
>
                     Nerd 
</option>,
                  <option
                    id="Girl Scout"
>
                     Girl Scout 
</option>,
                  <option
                    id="Shovel kid"
>
                     Shovel Kid
</option>,
                  <option
                    id="Collector"
>
                     Collector
</option>,
                  <option
                    id="Fat Kid"
>
                     Fat Kid
</option>,
                ],
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Pick a class ",
                  },
                  "ref": null,
                  "rendered": " Pick a class ",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Nerd ",
                    "id": "Nerd",
                  },
                  "ref": null,
                  "rendered": " Nerd ",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Girl Scout ",
                    "id": "Girl Scout",
                  },
                  "ref": null,
                  "rendered": " Girl Scout ",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Shovel Kid",
                    "id": "Shovel kid",
                  },
                  "ref": null,
                  "rendered": " Shovel Kid",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Collector",
                    "id": "Collector",
                  },
                  "ref": null,
                  "rendered": " Collector",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Fat Kid",
                    "id": "Fat Kid",
                  },
                  "ref": null,
                  "rendered": " Fat Kid",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  This is going to say something about the class you pick
</p>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This is going to say something about the class you pick",
                },
                "ref": null,
                "rendered": "This is going to say something about the class you pick",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " OK yea I made my guy ",
              },
              "ref": null,
              "rendered": " OK yea I made my guy ",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " here goes your hero gif",
          },
          "ref": null,
          "rendered": " here goes your hero gif",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
